contributors/design-proposals/storage-metrics.md

```diff
@@ -0,0 +1,91 @@
+# [WIP] Expose Storage Metrics to Users
+
+**Author**: Jing Xu (@jingxu97)
+
+**Last Updated**: 7/26/2017
+
+**Status**: Proposal
+
+This document explores different ways of exposing storage metrics to users
+
+## Background and Motivation
+
+Monitoring resource usage is critical for users/admins to manage their systems. For storage, there are three levels of metrics:
+ - Node-level: the capacity, used, and allocatable storage on local host
+ - Pod-level: Pod volume could be network attached (GCE PD, AWS EBS etc.) or local (emptyDir, hostPath, or local storage volume). Each volume has a total capacity and used bytes
+ - Container-level: How much storage is used by each container’s writable layer and logs (standard input/output).
+
+All the above metrics are used for operation of Kubernetes internal components and core utilities as explains below.
+
+ - Eviction manager checks the allocatable local storage (scratch space) and take actions. Scheduler also needs the node-level local storage capacity and availability to determine how to place pods on nodes 
+ - In 1.7, an alpha feature is added to isolation emptyDir volume usage and container’s overlay in local storage. User can set a sizeLimit for Pod EmptyDir Volume and eviction manager will take action if the emptyDir volume usage exceeds the configured size. Similarly, user can set request/limit on container’s overlay and eviction manager can take actions based on the monitoring data and the limits.
+ - There is also a proposal targeting for 1.8 [volume resize proposal](https://github.com/kubernetes/community/pull/657) to dynamically resize the volume (network attached) if the disk resource is insufficient. It requires a monitoring pipeline to dynamically check the the disk usage for each volume (represented by PVC)
+
+### Current Monitoring Options
+
+There are a number of monitoring options supported by Kubernetes
+ - Resource Metrics API (Alpha feature in progress [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md)
+ - Collect metrics in Heapster. (metrics has to exposed by Kubelet in Summary API)
+ - Export metrics through StackDriver 
+ - Kubectl Command (kubectl top, kubectl describe)
+
+All the storage metrics mentioned in Background are collected by Kubelet Summary API so that it could be integrated with the above monitoring system. However, pod volume metrics is provided in PodStat object’s VolumeStat and VolumeStat only has volume name specified in pod specification. There is no direct way of checking volume usage information by PVC name, which is more preferable by users.
+
+
+## Goals
+The gola of this proposal is to expose storage usage metrics to users for monitoring their storage systems.
```

Comment:
Is this feature on pod-level ?

Reply:
I think the metric is per volume. Could you explain what you mean by "pod-level"?

Reply:
spell: `s/gola/goal/`